controllers/hpa_util.go

```diff
@@ -127,10 +127,14 @@ func getHpaFromHvpa(hvpa *autoscalingv1alpha1.Hvpa) (*autoscaling.HorizontalPodA
 	return &autoscaling.HorizontalPodAutoscaler{
 		ObjectMeta: *metadata,
 		Spec: autoscaling.HorizontalPodAutoscalerSpec{
-			MaxReplicas:    hvpa.Spec.Hpa.Template.Spec.MaxReplicas,
-			MinReplicas:    hvpa.Spec.Hpa.Template.Spec.MinReplicas,
-			ScaleTargetRef: *hvpa.Spec.TargetRef.DeepCopy(),
-			Metrics:        hvpa.Spec.Hpa.Template.Spec.Metrics,
+			MaxReplicas: hvpa.Spec.Hpa.Template.Spec.MaxReplicas,
+			MinReplicas: hvpa.Spec.Hpa.Template.Spec.MinReplicas,
+			ScaleTargetRef: autoscaling.CrossVersionObjectReference{
+				APIVersion: "autoscaling.k8s.io/v1alpha1",
+				Kind:       "Hvpa",
+				Name:       hvpa.Name,
+			},
+			Metrics: hvpa.Spec.Hpa.Template.Spec.Metrics,
```

Comment:
DeepCopy?

Reply:
the hvpa instance passed here is anyway a local instance, and doesn't point to the cache. So didn't bother about deep-copying. wdyt?

Reply:
Might be safer to DeepCopy because it is now being pointed to from HPA and HVPA. Just to be safe.